server/routes/viewingLicences/handlers/printLicence.ts

```diff
@@ -14,67 +16,75 @@ const pdfHeaderFooterStyle =
   'padding: 20px;'
 
 export default class PrintLicenceRoutes {
-  constructor(private readonly prisonerService: PrisonerService) {}
+  constructor(private readonly prisonerService: PrisonerService, private readonly qrCodeService: QrCodeService) {}
 
   preview = async (req: Request, res: Response): Promise<void> => {
     const { username } = res.locals.user
-    const { licence } = res.locals
+    const { licence } = res.locals // fetchLicence middleware populates
+    const { qrCodesEnabled } = res.locals
     const htmlPrint = true
+
+    const qrCode = qrCodesEnabled ? await this.qrCodeService.getQrCode(licence) : null
     const additionalLicenceConditions = expandAdditionalConditions(licence.additionalLicenceConditions)
     const additionalPssConditions = expandAdditionalConditions(licence.additionalPssConditions)
+
+    logger.info(`HTML preview licence ID [${licence.id}] type [${licence.typeCode}] by user [${username}]`)
+
+    res.render(`pages/licence/${licence.typeCode}`, {
+      additionalLicenceConditions,
+      additionalPssConditions,
+      qrCodesEnabled,
```

Comment:
Since this is in `res.locals` it will already be in the view context, no need to provide it here

Reply:
oops, sorry I see its being changed here